Nodejs/Product/Telemetry/Shared/ITelemetryLogger.cs

```diff
@@ -0,0 +1,55 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+
+namespace Microsoft.NodejsTools.Telemetry {
+    public interface ITelemetryLogger : IDisposable {
+        /// <summary>
+        /// Records event with name
+        /// </summary>
+        ///<param name="eventName">Event name</param>
+        void ReportEvent(string eventName);
+
+        /// <summary>
+        /// Records event with single property
+        /// </summary>
+        /// <param name="eventName">Event Name</param>
+        /// <param name="propertyName">Property Name</param>
+        /// <param name="propertyValue">Property Value</param>
+        /// <param name="isPersonallyIdentifiable">Whether property has personally identifiable information</param>
+        void ReportEvent(string eventName, string propertyName, object propertyValue, bool isPersonallyIdentifiable = false);

```

Comment:
This will make it convenient (without adding another override to each implementation of this interface) for callers which use this interface as the compile time type.